docs/source/formulations_example.rst

```diff
@@ -0,0 +1,648 @@
+        predictor=ingredient_fractions_predictor, inputs=[flat_formulation],
+    )
+    price_descriptors = project.descriptors.from_predictor_responses(
+        predictor=mean_price_predictor, inputs=[flat_formulation],
+    )
+    sweetness_descriptors = project.descriptors.from_predictor_responses(
+        predictor=mean_sweetness_predictor, inputs=[flat_formulation]
+    )
+
+    blend_time = RealDescriptor("margarita~blend time", lower_bound=0, upper_bound=60, units="s")
+    tastiness = RealDescriptor("margarita~tastiness", lower_bound=0, upper_bound=10, units="")
+
+    ml_inputs = (
+        ingredient_fractions_descriptors
+        + label_fractions_descriptors
+        + price_descriptors
+        + sweetness_descriptors
+        + [blend_time]
+    )
+    ml_model = AutoMLPredictor(
+        name="ML model for tastiness",
+        description="",
+        output=tastiness,
+        inputs=ml_inputs
+    )
+
+Where did the descriptor keys ``margarita~blend time`` and ``margarita~tastiness`` come from?
+They came from concatenating the headers in the variables in the table, and the bounds and units came from the attribute templates.
+It's a lot to keep track of, we there is there :func:`~citrine.resources.descriptors.DescriptorMethods.descriptors_from_data_source` method.
+Calling ``project.descriptors.descriptors_from_data_source(data_source)`` returns a list of all of the descriptors emitted by the data source.
+Make sure that these are the descriptors you are using as inputs to your predictor.
+
+
+Lastly, we wrap everything in a :class:`~citrine.informatics.predictors.graph_predictor.GraphPredictor` and register it.
+
+.. code-block:: python
+
+    from citrine.informatics.predictors import GraphPredictor
+
+    predictors_list = [
+        simple_mixture_predictor,
+        ingredient_fractions_predictor,
+        label_fractions_predictor,
+        mean_price_predictor,
+        mean_sweetness_predictor,
+        ml_model
+    ]
+
+    graph_predictor = GraphPredictor(
+        name="Graphical model for tastiness of blended margaritas",
+        description="",
+        predictors=predictors_list,
+        training_data=[data_source] # the data source we defined above
+    )
+    graph_predictor = project.predictors.register(graph_predictor)
+
+One representation of this graphical model is shown below.
+
+.. figure:: _static/formulations_graphical_model.png
+    :align: center
+
+    Graphical model to predict tastiness of a margarita
+
+Defining a Design Space
+-----------------------
+
+Now that we have a trained predictor, the next step to finding the *world's best margarita* is to define the search space.
+A :class:`~citrine.informatics.design_spaces.formulation_design_space.FormulationDesignSpace` defines the ingredients, the labels, and any constraints.
+In addition to the labels used in the predictor, we also create a label for "simple syrup A" and "simple syrup B."
```

Comment:
MAYBE - Should we specify that this will only work if there are Ingredients with these labels already in the MH? I could see a lot of people getting to this stage and saying "Oh! I want to apply a new label. Why isn't it working?"

Reply:
This actually does not require that the labels exist in the material history. We are labeling the formulations emitted by the design space and constraining them directly, so there's no interaction with the predictor or the training data.